rasterio/session.py

```diff
@@ -316,6 +316,38 @@ def get_credential_options(self):
 
         """
         return {k.upper(): v for k, v in self.credentials.items()}
+    
+
+class GSSession(OSSSession):
+    """Configures access to secured resources stored in Google Cloud Storage
+    """
+    def __init__(self, google_application_credentials):
+        """Create new Google Cloude Storage session
+
+        Parameters
+        ----------
+        google_application_credentials: string
+            Path to the google application credentials JSON file.
+        """
 
+        self._creds = {
+            "google_application_credentials": google_application_credentials
+        }
```

Comment:
Let's insert a "google_application_credentials" item only if `google_application_credentials is not None`.